src/v/transform/logging/tests/log_manager_test.cc

```diff
@@ -0,0 +1,355 @@
+        _client = sink.get();
+        _manager = std::make_unique<manager_t>(
+          model::node_id{0},
+          std::move(sink),
+          _buffer_limit(),
+          _line_limit.bind(),
+          _flush_ms.bind());
+        start_manager();
+    }
+
+    void TearDown() override {
+        _client = nullptr;
+        _manager->stop().get();
+        _manager.reset();
+        set_buffer_limit(default_bl);
+        set_line_limit(default_ll);
+        set_flush_interval(default_flush);
+    }
+
+    void start_manager() { _manager->start().get(); }
+    void stop_manager() { _manager->stop().get(); }
+
+    void enqueue_log(
+      ss::log_level lvl,
+      model::transform_name_view name,
+      std::string_view msg) {
+        _manager->enqueue_log(lvl, name, msg);
+    }
+
+    void enqueue_log(std::string_view msg) {
+        enqueue_log(
+          ss::log_level::info,
+          model::transform_name_view{testing::random_transform_name()()},
+          msg);
+    }
+
+    const ss::chunked_fifo<iobuf>& logs() const { return _client->logs(); }
+
+    std::chrono::milliseconds flush_interval() { return _flush_ms(); }
+
+    void advance_clock(std::optional<ss::manual_clock::duration> dur = {}) {
+        ss::manual_clock::advance(dur.value_or(flush_interval()));
+        tests::drain_task_queue().get();
+    }
+
+    ss::sstring last_log_msg() {
+        if (logs().empty()) {
+            return ss::sstring{};
+        }
+        return testing::get_message_body(logs().back().copy());
+    }
+
+    void set_buffer_limit(size_t bl) { _buffer_limit.update(std::move(bl)); }
+    void set_line_limit(size_t ll) { _line_limit.update(std::move(ll)); }
+    void set_flush_interval(std::chrono::milliseconds fi) {
+        _flush_ms.update(std::move(fi));
+    }
+
+private:
+    fake_client* _client{};
+    std::unique_ptr<manager_t> _manager{};
+
+    static constexpr size_t default_bl{100_KiB};
+    config::mock_property<size_t> _buffer_limit{default_bl};
+
+    static constexpr size_t default_ll{1_KiB};
+    config::mock_property<size_t> _line_limit{default_ll};
+
+    static constexpr std::chrono::milliseconds default_flush{500ms};
+    config::mock_property<std::chrono::milliseconds> _flush_ms{default_flush};
+};
+
+TEST_F(TransformLogManagerTest, EnqueueLogs) {
+    enqueue_log("Hello from some test code!");
+
+    advance_clock();
+    EXPECT_EQ(logs().size(), 1);
+
+    enqueue_log("Hello again from some test code!");
+
+    advance_clock();
+    EXPECT_EQ(logs().size(), 2);
+
+    EXPECT_THAT(last_log_msg(), ::testing::HasSubstr("again"));
+}
+
+TEST_F(TransformLogManagerTest, LogsDroppedAtShutdown) {
+    constexpr size_t n = 10;
+
+    for (int i = 0; i < n; ++i) {
+        enqueue_log("Hello, World!");
+    }
+
+    EXPECT_EQ(logs().size(), 0);
+    stop_manager();
+    EXPECT_EQ(logs().size(), 0);
+}
+
+TEST_F(TransformLogManagerTest, LargeBuffer) {
+    // This will cause a reactor stall in Debug mode but NOT
+    // in Release
```

Comment:
Where is this from? Is that because `enqueue_log` is called multiple times synchronously? There isn't anything inherently in the manager that should cause stalls right? This is just an artifact of the test and Debug mode slowness?

Reply:
>isn't anything inherently in the manager that should cause stalls right?

correct. doesn't look like it to me, anyway.

>just an artifact of the test and debug mode slowness

I think so. It definitely stalls in `enqueue_log`, but it's a sparse backtrace with some asan symbols and some fmtlib (???) symbols in. I was more confident about this yesterday...backtrace decoding seems like it could be wrong.